packages/google-pay-integration/src/google-pay-button-strategy.ts

```diff
@@ -170,7 +170,7 @@ export default class GooglePayButtonStrategy implements CheckoutButtonStrategy {
             await this._paymentIntegrationService.updateBillingAddress(billingAddress);
         }
 
-        if (shippingAddress) {
+        if (shippingAddress && !this._buyNowCart?.lineItems.digitalItems.length) {
```

Comment:
Should updateShippingAddress being called if customer has both digital and physical items in the cart?

Reply:
Good point, missed this. I think in this case we will not call update address at all which leads to other issues.

Reply:
Good point. In this case it seems like we should call. I will double check

Reply:
But at the same time how is it possible?

Reply:
We work with one item on PDP

Reply:
I get it. We are talking about cart) I will double check

Reply:
I agree we are talking about PDP, but this button is rendered in cart summary and cart page as well. I think this is something @serhii-tkachenko and I discussed a while ago around adding different strategies for buy-now buttons. That should simplify the logic immensely in the long term. 
Something to think about.

Reply:
@serhii-tkachenko @animesh1987  In this case, everything works as expected. if we try to checkout from the cart with both items our checkout only works for the “Buy Now” feature, which means that if we have an update address, it will be updated successfully

Verified on dev store

Reply:
Oh I see, you are using this._buyNowCart in the condition... Should be ok then.